### PR TITLE
Rename pager methods. refs #3494

### LINKED ISSUE #3494: Wrong naming for Pager methods
Please, look at this:
https://github.com/sonata-project/SonataAdminBundle/blob/master/Datagrid/Pager.php#L254
https://github.com/sonata-project/SonataAdminBundle/blob/master/Datagrid/Pager.php#L268

Is it correct?? 
Maybe there: Pager::getFirstIndex() and Pager::getLastIndex()??

Thanks.

### PR DESCRIPTION
**Pull request by TimKam**
I am targeting this branch, because changing the names is a BC break.

Fixes #3494
## Changelog

``` markdown
### Changed
- Renamed pager methods `getFirstIndice()` and `getLastIndice()` to
 `getFirstIndex()`, respectively `getLastIndex()`.
```
## To do
- [X] Update the tests
## Subject

Rename the pager methods `getFirstIndice()` and `getLastIndice()`.

### CODE REVIEW

**Comment by greg0ire** [comment]
117 commits… this is not right. Please use `git rebase --interactive origin/master`, remove all commits that are not yours, then force push `git push --force`

**Comment by TimKam** [comment]
@greg0ire Thanks for the feedback. Can I already specify a minor version or should I keep it `3.x` in the deprecation warning?

**Comment by greg0ire** [comment]
I think you should stick to 3.x : we release quite often, and if your PR doesn't make it to the next release, it means you will have to rework it to bump the version numbers. Plus, it doesn't matter that much IMO.

**Comment by OskarStark** [comment]
shouldn't we deprecate the other methods and add an upgrade note or am I missing sth?

**Comment by greg0ire** [comment]
> shouldn't we deprecate the other methods

I think it goes without saying. And yes, you're right, an upgrade note will be needed.

File: Datagrid/Pager.php [hunk 1]
```diff
@@ -247,11 +247,27 @@ public function getPrevious()
     }
 
     /**
+     * NEXT_MAJOR: remove this method
+     *
+     * @deprecated
+     */
+
```
Comment (core23) [inline Datagrid/Pager.php]: Please remove the empty line here

File: Datagrid/Pager.php [hunk 2]
```diff
@@ -247,11 +247,27 @@ public function getPrevious()
     }
 
     /**
+     * NEXT_MAJOR: remove this method
+     *
+     * @deprecated
```
Comment (core23) [inline Datagrid/Pager.php]: Please add a message to the `@deprecated` annoation like we do for any other method

File: Datagrid/Pager.php [hunk 3]
```diff
@@ -260,12 +276,29 @@ public function getFirstIndice()
         return ($this->page - 1) * $this->maxPerPage + 1;
     }
 
+
+    /**
+     * NEXT_MAJOR: remove this method
+     *
+     * @deprecated
+     */
+
```
Comment (core23) [inline Datagrid/Pager.php]: Please remove the empty line here

File: Datagrid/Pager.php [hunk 4]
```diff
@@ -260,12 +276,29 @@ public function getFirstIndice()
         return ($this->page - 1) * $this->maxPerPage + 1;
     }
 
+
+    /**
+     * NEXT_MAJOR: remove this method
+     *
+     * @deprecated
```
Comment (core23) [inline Datagrid/Pager.php]: Please add a message to the `@deprecated` annoation like we do for any other method

**Comment by OskarStark** [comment]
@TimKam can you please fullfill the comments of @core23 ?

**Comment by OskarStark** [comment]
after this PR is merged, we need to fix the PagerInterface and remove the deprecations

**Comment by OskarStark** [comment]
I reworked this PR here: https://github.com/sonata-project/SonataAdminBundle/pull/4263

**Pull request closed without merging**